crates/driver/Cargo.toml

```diff
@@ -23,10 +23,11 @@ hyper = "0.14"
 serde_with = "2.0"
 async-trait = "0.1"
 itertools = "0.10"
-url = "2.3"
+url = { version = "2.3", features = ["serde"] }
 ethabi = "18.0"
 rand = "0.8"
 web3 = "0.18"
+serde_yaml = "0.9"
```

Comment:
TOML makes nested objects harder to represent. YAML is a little nicer for that, but the downside being that it is sensitive to whitespace and generally a much more complicated format.

I would suggest:
- If the configuration is mostly flat, TOML because it is simpler
- If the configuration will be deeply nested, YAML is more convenient.

Reply:
My **personal** preference is TOML, but mostly because it is much simpler than YAML ([TOML spec](https://toml.io/en/v1.0.0) is less then half the size of the [YAML spec](https://yaml.org/spec/1.2.2/)) and has [fewer gotchas](https://noyaml.com/).

That being said, I do see the value of YAML for complicated configurations.